apps/prairielearn/src/pages/instructorEffectiveUser/instructorEffectiveUser.html.ts

```diff
@@ -0,0 +1,345 @@
+  CoursePermissionSchema,
+  UserSchema,
+} from '../../lib/db-types.js';
+
+export const CourseRolesSchema = z.object({
+  available_course_roles: CoursePermissionSchema.shape.course_role.unwrap().array(),
+  available_course_instance_roles: CourseInstancePermissionSchema.shape.course_instance_role
+    .unwrap()
+    .array(),
+  available_uids: UserSchema.shape.uid.array().nullable(),
+});
+export type CourseRoles = z.infer<typeof CourseRolesSchema>;
+
+export function InstructorEffectiveUser({
+  resLocals,
+  ipAddress,
+  courseRoles,
+}: {
+  resLocals: Record<string, any>;
+  ipAddress: string;
+  courseRoles: CourseRoles;
+}) {
+  const { authz_data, course_instance, __csrf_token, req_date, true_req_date, user } = resLocals;
+
+  // This page can be mounted under `/pl/course/...`, in which case we won't
+  // have a course instance to get a display timezone from. In that case, we'll
+  // fall back to the course, and then to the institution. All institutions must
+  // have a display timezone, so we're always guaranteed to have one.
+  const displayTimezone =
+    resLocals.course_instance?.display_timezone ??
+    resLocals.course.display_timezone ??
+    resLocals.institution.display_timezone;
+
+  const formattedTrueReqDate = format(
+    toZonedTime(true_req_date, displayTimezone),
+    "yyyy-MM-dd'T'HH:mm:ss.SSSxxx",
+    { timeZone: displayTimezone },
+  );
+  const formattedReqDate = format(
+    toZonedTime(req_date, displayTimezone),
+    "yyyy-MM-dd'T'HH:mm:ss.SSSxxx",
+    { timeZone: displayTimezone },
+  );
+
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(import.meta.url, "<%- include('../partials/head'); %>", {
+          ...resLocals,
+          pageTitle: 'Change Effective User',
+        })}
+      </head>
+      <body>
+        ${renderEjs(import.meta.url, "<%- include('../partials/navbar'); %>", {
+          ...resLocals,
+          navPage: 'effective',
+        })}
+        <main id="content" class="container-fluid">
+          <div class="card mb-4">
+            <div class="card-header bg-primary text-white">Authenticated user</div>
+
+            <div class="card-body">
+              <p><strong>Authenticated UID:</strong> ${authz_data.authn_user.uid}</p>
+              <p><strong>Authenticated name:</strong> ${authz_data.authn_user.name}</p>
+              <p><strong>Authenticated course role:</strong> ${authz_data.authn_course_role}</p>
+              ${course_instance
+                ? html`
+                    <p>
+                      <strong>Authenticated course instance role:</strong>
+                      ${authz_data.authn_course_instance_role}
+                    </p>
+                  `
+                : ''}
+              <p><strong>Authenticated mode:</strong> ${authz_data.authn_mode}</p>
+              <p><strong>Authenticated date:</strong> ${formattedTrueReqDate}</p>
+
+              <form id="resetForm" method="POST">
+                <p>
```

Comment:
If reviewer is comparing with original EJS: this was reordered. The original had the form inside the paragraph, but that does not provide the expected result, as the start of `form` [causes `p` to be presumed closed](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/p#technical_summary). In other words, `<p><form>...</form></p>` is equivalent to `<p></p><form>...</form>`.